stateproof/builder.go

```diff
@@ -34,6 +34,34 @@ import (
 	"github.com/algorand/go-algorand/stateproof/verify"
 )
 
+// OnPrepareCommit is a function called by the ledger when it's preparing to commit rnd. It gives the builder
+// the chance to persist the data it needs.
+func (spw *Worker) OnPrepareCommit(rnd basics.Round) {
```

Comment:
Just to make sure, is it really OK to reference the ledger in a callback like this?